client/src/pages/spools/edit.tsx

```diff
@@ -35,14 +35,60 @@ export const SpoolEdit: React.FC<IResourceComponentsProps> = () => {
     return {
       label: label,
       value: item.id,
+      weight: item.weight,
+      spool_weight: item.spool_weight,
     };
   });
   filamentOptions?.sort((a, b) => a.label.localeCompare(b.label, undefined, { sensitivity: "base" }));
 
+  const [weightToEnter, setWeightToEnter] = useState(1);
+  const [usedWeight, setUsedWeight] = useState(0);
+
+  const selectedFilamentID = Form.useWatch("filament_id", form);
+  const selectedFilament = filamentOptions?.find((obj) => {
+    return obj.value === selectedFilamentID;
+  });
+  const filamentWeight = selectedFilament?.weight || 0;
+  const spoolWeight = selectedFilament?.spool_weight || 0;
+
+  const filamentChange = (newID: number) => {
+    const newSelectedFilament = filamentOptions?.find((obj) => {
+      return obj.value === newID;
+    });
+    const newFilamentWeight = newSelectedFilament?.weight || 0;
+    const newSpoolWeight = newSelectedFilament?.spool_weight || 0;
+
+    if (weightToEnter >= 3) {
+      if (!(newFilamentWeight && newSpoolWeight)) {
+        setWeightToEnter(2);
+      }
+    }
+    if (weightToEnter >= 2) {
+      if (!newFilamentWeight) {
+        setWeightToEnter(1);
+      }
+    }
+  };
+
+  const weightChange = (weight: number) => {
+    setUsedWeight(weight);
+    form.setFieldsValue({
+      used_weight: weight,
+    });
+  };
+
   if (formProps.initialValues) {
     formProps.initialValues["filament_id"] = formProps.initialValues["filament"].id;
   }
 
+  useEffect(() => {
+    if (formProps.initialValues && usedWeight != formProps.initialValues["used_weight"]) {
+      setUsedWeight(formProps.initialValues["used_weight"])
+    }
+  }, []);
+
+  console.log(usedWeight)
```

Comment:
Remove

Reply:
just the console.log right? not the useEffect part?

Reply:
Yeah correct